server/modlog.ts

```diff
@@ -497,59 +500,60 @@ export class Modlog {
 		if (rooms.includes('global')) roomChecker = `(roomid LIKE 'global-%' OR ${roomChecker})`;
 
 		let query = `SELECT *, (SELECT group_concat(userid, ',') FROM alts WHERE alts.modlog_id = modlog.modlog_id) as alts FROM modlog`;
+		if (this.useFTS) query += ` LEFT JOIN modlog_fts ON modlog_fts.rowid = modlog_id`;
 		query += ` WHERE ${roomChecker}`;
 
 		if (search.anyField) {
-			query += ` AND action LIKE '%' || ? || '%'`;
+			query += ` AND action LIKE ? || '%'`;
 
-			query += ` OR userid LIKE '%' || ? || '%'`;
-			query += ` OR autoconfirmed_userid LIKE '%' || ? || '%'`;
-			query += ` OR EXISTS(SELECT * FROM alts WHERE alts.modlog_id = modlog.modlog_id AND userid LIKE '%' || ? || '%')`;
+			query += ` OR userid LIKE ? || '%'`;
+			query += ` OR autoconfirmed_userid LIKE ? || '%'`;
+			query += ` OR EXISTS(SELECT * FROM alts WHERE alts.modlog_id = modlog.modlog_id AND userid LIKE ? || '%')`;
 
-			query += ` OR ip LIKE '%' || ? || '%'`;
-			query += ` OR action_taker_userid LIKE '%' || ? || '%'`;
+			query += ` OR ip LIKE ? || '%'`;
+			query += ` OR action_taker_userid LIKE ? || '%'`;
 
 			args.push(...Array(6).fill(search.anyField));
 
 			if (this.useFTS) {
-				query += ` OR note MATCH ?`;
+				query += ` OR modlog_fts.note MATCH ?`;
 				args.push(toID(search.anyField));
 			} else {
-				query += ` OR note LIKE '%' || ? || '%'`;
+				query += ` OR note LIKE ? || '%'`;
```

Comment:
We shouldn't have an index on note anyway, so this will always be `O(n)` for non-FTS case.

Would it be simpler if we just always required FTS, but used the default tokenization (word boundaries) for when you don't want to build the native extension?

Reply:
Yeah, that would be simpler.